pkg/config/release.go

```diff
@@ -159,6 +169,10 @@ func GetAllConfigsFromSHA(releaseRepoPath, sha string) (*ReleaseRepoConfig, erro
 	if restoreRev == "HEAD" {
 		restoreRev = currentSHA
 	}
+	// git fetch to be sure we are in sync
```

Comment:
Shouldn't this be done in the `pj-rehearse` code when we process a command?  Based on a quick `grep`, this function is also used by `config-change-trigger`, and it is unexpected that it now suddenly potentially performs a network request.

Reply:
In addition, we load the base configuration after the candidate, so I don't see how we could not have already fetched the base branch.

Reply:
> Shouldn't this be done in the pj-rehearse code when we process a command? Based on a quick grep, this function is also used by config-change-trigger, and it is unexpected that it now suddenly potentially performs a network request.

That's a fair point. It looks like this issue hasn't occurred in the last couple of days anyways, so I am going to hold this for now. If we do see the problem again we can find a more specific way to do this.

> In addition, we load the base configuration after the candidate, so I don't see how we could not have already fetched the base branch.

The error is definitely puzzling. This was really just something to try based on that stack overflow post. There is no deterministic way that I have discovered to reproduce this, so there isn't much better I can do right now. If it occurs as infrequently as it is now, I don't think we need to worry about it too much.